pkg/cluster/cluster.go

```diff
@@ -782,15 +762,31 @@ func (c *Cluster) Update(oldSpec, newSpec *acidv1.Postgresql) error {
 	}
 
 	// sync connection pooler
-	if _, err := c.syncConnectionPooler(oldSpec, newSpec,
-		c.installLookupFunction); err != nil {
-		c.logger.Errorf("could not sync connection pooler: %v", err)
-		updateFailed = true
+	for _, role := range c.RolesConnectionPooler() {
+		if _, err := c.ConnectionPooler[role].syncConnectionPooler(oldSpec, newSpec,
+			c.installLookupFunction); err != nil {
+			c.logger.Errorf("could not sync connection pooler: %v", err)
+			updateFailed = true
+		}
 	}
 
 	return nil
 }
 
+func syncResources(a, b *v1.ResourceRequirements) bool {
```

Comment:
I understand this function was named like that in the original code, but judging from what it does, `shouldSyncResources` or `needToSyncResouces` is a better name for it from a readability perspective.